### ISSUE TITLE
Inconsistency between the pretrained models and labels

### ISSUE THREAD

**Issue by seyeeet** [pytorch/vision#3429]
I notice that for pretrain models that are provided the labels are not consistent.
For example vgg16 class 1 is different from Resnet50 class 1.
Can you let us know where we can find the corresponding labels for each model?
For vgg i notice that the one that looks like this:
```{
  "0": [
    "n01440764",
    "tench"
  ],
  "1": [
    "n01443537",
    "goldfish"
  ],
  "2": [
    "n01484850",
    "great_white_shark"
  ],
  "3": [
    "n01491361",
    "tiger_shark"
  ],
  "4": [
    "n01494475",
    "hammerhead"
  ],
  "5": [
    "n01496331",
    "electric_ray"
  ],
  "6": [
    "n01498041",
    "stingray"
  ],
  "7": [
    "n01514668",
    "cock"
  ],
  "8": [
    "n01514859",
    "hen"
```
works, but this one is not the one that we should use for resnets, please let us know what we should do.
Thanks

**Comment by seyeeet** [comment]
any update on this @pmeier

**Comment by fmassa** [comment]
Hi,

I believe all classification models in torchvision have the same expected output order (the order following the sorted imagenet synsets) https://github.com/pytorch/vision/blob/c991db82abba12e664eeac14c9b643d0f1f1a7df/torchvision/datasets/folder.py#L164-L165

We can verify that because we have recently re-computed all model accuracies in https://github.com/pytorch/vision/pull/3360, using the same dataset for all models (so the class indices are the same).

As such, I believe there shouldn't be any inconsistency there, and the behavior you might have seen is because VGG16 misclassified one image.

I believe I've answered your question and as such I'm closing this issue, but let us know if you have any further questions

**Comment by seyeeet** [comment]
@fmassa thanks for your reply, but I am not sure if that is solved.
I still cannot achieve a reasonable accuracy via pretrained models. 
To be specific, in the txt file for labels I have:

```
n01978287 613 Dungeness_crab 
n01978455 614 rock_crab
n01980166 615 fiddler_crab
n01981276 616 king_crab
...
```
can you clarify how should I sort the labels? currently, I am sorting based on the first elements (i.e. I'm sorting based on n0... class numbers), but this result is very bad.

should I sort based on class ids (i.d. second elements in the line)? 
Thanks for your help

**Comment by fmassa** [comment]
Hi,

Here is a mapping between the output idxs and the class names https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt

You can see that the label names have been alphabetically ordered by synset in https://gist.github.com/fnielsen/4a5c94eaa6dcdf29b7a62d886f540372
